### ISSUE TITLE
`npx create-expo-app@latest` is failing with link-react-compiler-runtime.sh: not found

### ISSUE THREAD

**Issue by ecruzolivera** [expo/expo#31912]
### Summary

Suddenly (2024-10-07), if I run `npx create-expo-app@latest TestExpo` the install process gets forever and finally fails with:

### Using npm

```
pm error code 127
npm error path /home/ernesto/Scratchpad/TestExpo/node_modules/babel-plugin-react-compiler
npm error command failed
npm error command sh -c ./scripts/link-react-compiler-runtime.sh
npm error sh: 1: ./scripts/link-react-compiler-runtime.sh: not found
npm error A complete log of this run can be found in: /home/ernesto/.npm/_logs/2024-10-08T01_05_52_912Z-debug-0.log
Something went wrong installing JavaScript dependencies. Check your npm logs. Continuing to create the app.
Error: npm install exited with non-zero code: 127

```
Full logs:

[2024-10-08T01_05_52_912Z-debug-0.log](https://github.com/user-attachments/files/17286221/2024-10-08T01_05_52_912Z-debug-0.log)

### Using yarn

```
[4/4] Building fresh packages...
verbose 32.140710131 Error: /home/ernesto/Scratchpad/TestExpo/node_modules/babel-plugin-react-compiler: Command failed.
Exit code: 127
Command: ./scripts/link-react-compiler-runtime.sh
Arguments:
Directory: /home/ernesto/Scratchpad/TestExpo/node_modules/babel-plugin-react-compiler
Output:
/bin/sh: 1: ./scripts/link-react-compiler-runtime.sh: not found
    at ProcessTermError.ExtendableBuiltin (/home/ernesto/.local/share/mise/installs/node/20.17.0/lib/node_modules/yarn/lib/cli.js:721:66)
    at ProcessTermError.MessageError (/home/ernesto/.local/share/mise/installs/node/20.17.0/lib/node_modules/yarn/lib/cli.js:750:123)
    at new ProcessTermError (/home/ernesto/.local/share/mise/installs/node/20.17.0/lib/node_modules/yarn/lib/cli.js:790:113)
    at ChildProcess.<anonymous> (/home/ernesto/.local/share/mise/installs/node/20.17.0/lib/node_modules/yarn/lib/cli.js:25789:17)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:519:28)
    at Pipe.<anonymous> (node:net:339:12)
error /home/ernesto/Scratchpad/TestExpo/node_modules/babel-plugin-react-compiler: Command failed.
Exit code: 127
Command: ./scripts/link-react-compiler-runtime.sh
Arguments:
Directory: /home/ernesto/Scratchpad/TestExpo/node_modules/babel-plugin-react-compiler
Output:
/bin/sh: 1: ./scripts/link-react-compiler-runtime.sh: not found
```

- OS Ubuntu24.04
- Linux 6.8.0-45-generic
- node v20.17.0 installed via [mise](https://mise.jdx.dev/)
- npm  v10.9.0

Yesterday I was able to create a project just fine

### SDK Version

latest

### Environment

```text
expo-env-info 1.2.0 environment info:
    System:
      OS: Linux 6.8 Ubuntu 24.04.1 LTS 24.04.1 LTS (Noble Numbat)
      Shell: 5.9 - /usr/bin/zsh
    Binaries:
      Node: 20.17.0 - ~/.local/share/mise/installs/node/lts/bin/node
      Yarn: 1.22.22 - ~/.local/share/mise/installs/node/lts/bin/yarn
      npm: 10.9.0 - ~/.local/share/mise/installs/node/lts/bin/npm
      Watchman: 4.9.0 - /usr/bin/watchman
    npmGlobalPackages:
      eas-cli: 12.5.1
    Expo Workflow: managed
```
```


### Minimal reproducible example

Just run `npx create-expo-app@latest Example`

**Comment by SkorpionG** [comment]
I have the same issue here. Initially, I thought the problem was in my npm, but npm works fine with my other react project.

**Comment by ddikodroid** [comment]
i'm getting the same issue
<img width="645" alt="Screenshot 2024-10-08 at 09 34 14" src="https://github.com/user-attachments/assets/6b52a9e3-479e-4a5b-8743-c94e41494922">

**Comment by RazvanGorea** [comment]
+ 1

**Comment by caltamiranob** [comment]
i have the same issue

**Comment by nathangemtech** [comment]
I am getting same error

**Comment by Louis-Westmoreland** [comment]
same

**Comment by xuoutput** [comment]
same

**Comment by rosko** [comment]
Same here as well
```
npm error code 127
npm error path /Users/rosko/Projects/my-app/node_modules/babel-plugin-react-compiler
npm error command failed
npm error command sh -c ./scripts/link-react-compiler-runtime.sh
npm error sh: ./scripts/link-react-compiler-runtime.sh: No such file or directory
npm error A complete log of this run can be found in: /Users/rosko/.npm/_logs/2024-10-08T03_04_39_259Z-debug-0.log
Something went wrong installing JavaScript dependencies. Check your npm logs. Continuing to create the app.
Error: npm install exited with non-zero code: 127

✅ Your project is ready!
```

Log:
```
6443 info run fsevents@2.3.3 install node_modules/fsevents node-gyp rebuild
6444 info run fsevents@2.3.3 install { code: 1, signal: null }
6445 verbose reify failed optional dependency /Users/rosko/Projects/my-app/node_modules/fsevents
6446 silly reify mark deleted [ '/Users/rosko/Projects/my-app/node_modules/fsevents' ]
6447 info run babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007 postinstall node_modules/babel-plugin-react-compiler ./scripts/link-react-compiler-runtime.sh
6448 info run babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007 postinstall { code: 127, signal: null }
6449 verbose stack Error: command failed
6449 verbose stack     at promiseSpawn (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
6449 verbose stack     at spawnWithShell (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
6449 verbose stack     at promiseSpawn (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
6449 verbose stack     at runScriptPkg (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:77:13)
6449 verbose stack     at runScript (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
6449 verbose stack     at /Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/rebuild.js:329:17
6449 verbose stack     at run (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:67:22)
6449 verbose stack     at /Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:84:9
6449 verbose stack     at new Promise (<anonymous>)
6449 verbose stack     at callLimit (/Users/rosko/.nvm/versions/node/v20.17.0/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:35:69)
6450 verbose pkgid babel-plugin-react-compiler@0.0.0-experimental-7779988-20241007
6451 error code 127
6452 error path /Users/rosko/Projects/my-app/node_modules/babel-plugin-react-compiler
6453 error command failed
6454 error command sh -c ./scripts/link-react-compiler-runtime.sh
6455 error sh: ./scripts/link-react-compiler-runtime.sh: No such file or directory
6456 silly unfinished npm timer reify 1728356679586
6457 silly unfinished npm timer reify:build 1728356704220
6458 silly unfinished npm timer build 1728356704222
6459 silly unfinished npm timer build:deps 1728356704223
6460 silly unfinished npm timer build:run:postinstall 1728356705180
6461 silly unfinished npm timer build:run:postinstall:node_modules/babel-plugin-react-compiler 1728356705180
6462 verbose cwd /Users/rosko/Projects/my-app
6463 verbose os Darwin 23.6.0
6464 verbose node v20.17.0
6465 verbose npm  v10.8.2
6466 verbose exit 127
6467 verbose code 127
```

**Comment by rosko** [comment]
As a temporary solution, I added the following to `devDependencies` in `package.json`
```
"babel-plugin-react-compiler": "0.0.0-experimental-592953e-20240517"
```

**Comment by ravindUwU** [comment]
I'm not using the React compiler just yet, so as an interim fix I was able to override the version of `babel-plugin-react-compiler` to `0.0.0-experimental-734b737-20241003` (which is the version that immediately precedes the latest version as of right now, listed at https://www.npmjs.com/package/babel-plugin-react-compiler?activeTab=versions), and have it install successfully.

**Comment by ddikodroid** [comment]
> As a temporary solution, I added the following to `devDependencies` in `package.json`
> 
> ```
> "babel-plugin-react-compiler": "0.0.0-experimental-592953e-20240517"
> ```

i'm still getting the error on fresh project

```sh
error /Users/diko/Documents/Project/risk-management/node_modules/babel-preset-expo/node_modules/babel-plugin-react-compiler: Command failed.
Exit code: 127
Command: ./scripts/link-react-compiler-runtime.sh
Arguments: 
Directory: /Users/diko/Documents/Project/risk-management/node_modules/babel-preset-expo/node_modules/babel-plugin-react-compiler
Output:
/bin/sh: ./scripts/link-react-compiler-runtime.sh: No such file or directory

```

**Comment by treymorgan** [comment]
same issue.

**Comment by CallbackDan** [comment]
same

**Comment by TowhidKashem** [comment]
this will let it build successfully in the mean time:

```
  "resolutions": {
    "**/babel-plugin-react-compiler": "0.0.0-experimental-ca8e0be-20240916"
  },
  ```

but only if you already have a `yarn.lock` file. If you don't (such as in CI) it will still fail to build.

**Comment by vitor3g** [comment]
same

**Comment by f1rstf1y9** [comment]
same issue😞

**Comment by BoChingLee** [comment]
same issue...

**Comment by snorlaxbubba** [comment]
oh my god i'm so happy to have come across this, I literally did a build 24 hours ago that worked fine

**Comment by basememara** [comment]
I tried `"babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"` but didn't work for me. any other workarounds or what expo version should we try?

**Comment by SagarAhir** [comment]
facing same issue

**Comment by basememara** [comment]
I tried going back a few versions back for expo and even babel, but seems like the internal dependency is still broken. completely blocked from any new expo project

**Comment by tuanlongn** [comment]
@basememara You can try using `yarn install --ignore-scripts` as a workaround.

**Comment by basememara** [comment]
Thx that did get me past the install and not sure what the implications of that is. However, when I try to do anything else after that, like `npx expo install expo-dev-client`, it gives this error:

```
error /Users/*/node_modules/@babel/core: Command failed.
Exit code: 1
Command: node scripts/postinstall.js
Arguments: 
Directory: /Users/*/node_modules/@babel/core
Output:
node:internal/modules/esm/resolve:844
  throw new ERR_MODULE_NOT_FOUND(packageName, fileURLToPath(base), null);
        ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'is-ci' imported from /Users/*/node_modules/@babel/core/scripts/postinstall.js
    at packageResolve (node:internal/modules/esm/resolve:844:9)
    at moduleResolve (node:internal/modules/esm/resolve:901:20)
    at defaultResolve (node:internal/modules/esm/resolve:1121:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.10.0
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
Error: yarnpkg add expo-dev-client@~4.0.27 exited with non-zero code: 1
Error: yarnpkg add expo-dev-client@~4.0.27 exited with non-zero code: 1
    at ChildProcess.completionListener (/Users/*/node_modules/@expo/spawn-async/build/spawnAsync.js:42:23)
    at Object.onceWrapper (node:events:629:26)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
    ...
    at spawnAsync (/Users/*/node_modules/@expo/spawn-async/build/spawnAsync.js:7:23)
    at YarnPackageManager.runAsync (/Users/*/node_modules/@expo/package-manager/build/node/BasePackageManager.js:41:42)
    at /Users/*/node_modules/@expo/package-manager/build/node/YarnPackageManager.js:35:127
    at /Users/*/node_modules/@expo/package-manager/build/utils/spawn.js:19:34
```

**Comment by Punith-goud** [comment]
this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
npm warn deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated rimraf@2.6.3: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.1.6: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.1.6: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.1.6: Glob versions prior to v9 are no longer supported
npm warn cleanup Failed to remove some directories [
npm warn cleanup   [
npm warn cleanup     'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\eol',
npm warn cleanup     [Error: EBUSY: resource busy or locked, rmdir 'C:\zysk\reactNative\shubham-k-mart\node_modules\eol'] {
npm warn cleanup       errno: -4082,
npm warn cleanup       code: 'EBUSY',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\eol'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules',
npm warn cleanup     [Error: EBUSY: resource busy or locked, rmdir 'C:\zysk\reactNative\shubham-k-mart\node_modules\eol'] {
npm warn cleanup       errno: -4082,
npm warn cleanup       code: 'EBUSY',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\eol'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\@react-native\\community-cli-plugin\\node_modules\\@react-native',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\zysk\reactNative\shubham-k-mart\node_modules\@react-native\community-cli-plugin\node_modules\@react-native'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\@react-native\\community-cli-plugin\\node_modules\\@react-native'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\expo-router\\node_modules',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\zysk\reactNative\shubham-k-mart\node_modules\expo-router\node_modules'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\expo-router\\node_modules'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\expo-camera',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\zysk\reactNative\shubham-k-mart\node_modules\expo-camera\android\src'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\expo-camera\\android\\src'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\react-native',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\zysk\reactNative\shubham-k-mart\node_modules\react-native\ReactAndroid\src\main\java\com\facebook\react\internal\turbomodule'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\zysk\\reactNative\\shubham-k-mart\\node_modules\\react-native\\ReactAndroid\\src\\main\\java\\com\\facebook\\react\\internal\\turbomodule'
npm warn cleanup     }
npm warn cleanup   ]
npm warn cleanup ]
npm error code 1
npm error path C:\zysk\reactNative\shubham-k-mart\node_modules\babel-plugin-react-compiler
npm error command failed
npm error command C:\Windows\system32\cmd.exe /d /s /c ./scripts/link-react-compiler-runtime.sh
npm error '.' is not recognized as an internal or external command,
npm error operable program or batch file.
npm error A complete log of this run can be found in: C:\Users\PunithGoud\AppData\Local\npm-cache\_logs\2024-10-08T06_47_34_687Z-debug-0.log
PS C:\zysk\reactNative\shubham-k-mart> 


i dont know if its my issue or its broken, i dont know what to do.

**Comment by WebbkodsLarlingen** [comment]
Adding this to my `package.json`, then deleting node_modules and running `npm install` works for me:

`"overrides": {
    "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
  },`

This is on Windows 10 using just WSL (no WSL2 though) on latest version of VSCode. I did this AFTER running `npx create-expo-app@latest` and I will try now with blank project since I am just learning React Native Expo from scratch and I need a blank project to follow through on.

Thanks for all the help, fellas! ^_^

Best regards,
WKF.

**Comment by chan-stclab** [comment]
Thank you genius people

**Comment by chan-stclab** [comment]
@ravindUwU 
Despite [react repo PR](https://github.com/facebook/react/pull/31147) merged, it didn't solved yet.
 another [Issue](https://github.com/facebook/react/issues/31150) raised. keep watching

**Comment by Gabriel-Ivan5795** [comment]
Adding "babel-plugin-react-compiler": "0.0.0-experimental-592953e-20240517" in devDependencies seems works in my case

**Comment by mozart25** [comment]
same

**Comment by vlmcode** [comment]
same

**Comment by AndryanaBaros** [comment]
> Adding "babel-plugin-react-compiler": "0.0.0-experimental-592953e-20240517" in devDependencies seems works in my case

its not working for me . sad. :(

**Comment by erdemozkok** [comment]
same problem and `"overrides": { "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003" }` doesn't work in my case

**Comment by pixcai** [comment]
same

**Comment by hxzrx** [comment]
Same error.

**Comment by diyar730** [comment]
same error

**Comment by SharpAccount** [comment]
> same problem and `"overrides": { "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003" }` doesn't work in my case

try to change version from "0.0.0-experimental-734b737-20241003" to "0.0.0"

**Comment by diogotl** [comment]
sameeeee :(

**Comment by diwashTiwari** [comment]
I tried several ways, including uninstalling npm, updating it, and using Yarn, but nothing seemed to work. 

However, this one worked for me 

```
"overrides": {
  "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
}
```

**Comment by hungnguyenviethatinh** [comment]
`"resolutions": {
    "**/babel-plugin-react-compiler": "0.0.0-experimental-ca8e0be-20240916"
  },`

this worked, many thanks!

**Comment by theodrosyimer** [comment]
> I tried several ways, including uninstalling npm, updating it, and using Yarn, but nothing seemed to work.
> 
> However, this one worked for me
> 
> ```
> "overrides": {
>   "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
> }
> ```

This worked for me using npm but does not work using pnpm

**Comment by antenorq** [comment]
The "overrides": { "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003" } WORKS FOR ME!

But I am getting this error:
Starting project at C:\Users\anten\xxxxx
Starting Metro Bundler
Failed to resolve the Android SDK path. Default install location not found: C:\Users\anten\AppData\Local\Android\Sdk. Use ANDROID_HOME to set the Android SDK location.

**Comment by diwashTiwari** [comment]
> The "overrides": { "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003" } WORKS FOR ME!
> 
> But I am getting this error: Starting project at C:\Users\anten\xxxxx Starting Metro Bundler Failed to resolve the Android SDK path. Default install location not found: C:\Users\anten\AppData\Local\Android\Sdk. Use ANDROID_HOME to set the Android SDK location.

The error is caused by a missing Android SDK, either because it's not installed or the path is not set correctly. 
If the SDK is installed, the ANDROID_HOME environment variable may not be configured properly, and the default location (C:\Users\anten\AppData\Local\Android\Sdk) cannot be found.

**Comment by byCedric** [comment]
Hey all! This is caused by the `babel-plugin-react-compiler` package itself, they accidentally published a version that contains an incorrect `postinstall` script. See https://github.com/facebook/react/issues/31150

**Comment by byCedric** [comment]
For now, we will issue a hotfix for `babel-preset-expo`, as this is the package that pulls the react compiler into Expo projects.

You can also install with `npm install --ignore-scripts`, to avoid the problematic `postinstall` hook in the react compiler. Or, you could change the the installed version through resolutions/overrides. That should get you unblocked for now.

**Comment by mlazari** [comment]
> This worked for me using npm but did not work using pnpm

@theodrosyimer It looks like for pnpm you have to put the overrides under a "pnpm" key (https://pnpm.io/package_json#pnpmoverrides) like this:

```
"pnpm": {
  "overrides": {
    "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
  }
}
```

For yarn I think it would be:
```
  "resolutions": {
    "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
  }
```

**Comment by erdemozkok** [comment]
> For yarn I think it would be:
> 
> ```
>   "resolutions": {
>     "babel-plugin-react-compiler": "0.0.0-experimental-734b737-20241003"
>   }
> ```

works for me

**Comment by byCedric** [comment]
Hey all! We've [issued an emergency hotfix](https://github.com/expo/expo/pull/31917) for `babel-preset-expo@11.0.15`. This pins the React Compiler version to an older but working one. We've also pinged Meta to get a proper fix for `babel-plugin-react-compiler` ASAP.

For now, it's likely that using `experiments.reactCompiler = true` will break your project. But if you don't use the React Compiler experiment, your project will install normally.

If you still encounter this, update your lock file to install `babel-preset-expo@11.0.15` for SDK 51, not `babel-preset-expo@<=11.0.14`. New projects created with `npx create-expo(-app)` should work again.

Sorry for the trouble!

**Comment by reactWebztechieMD** [comment]
i am getting this error when i am making build

**Comment by diyar730** [comment]
> Hey all! We've [issued an emergency hotfix](https://github.com/expo/expo/pull/31917) for `babel-preset-expo@11.0.15`. This pins the React Compiler version to an older but working one. We've also pinged Meta to get a proper fix for `babel-plugin-react-compiler` ASAP.
> 
> For now, it's likely that using `experiments.reactCompiler = true` will break your project. But if you don't use the React Compiler experiment, your project will install normally.
> 
> If you still encounter this, update your lock file to install `babel-preset-expo@11.0.15` for SDK 51, not `babel-preset-expo@<=11.0.14`. New projects created with `npx create-expo(-app)` should work again.
> 
> Important
> 
> `babel-preset-expo` is a dependency of `expo` and, preferably, should not be installed directly in your project. However, you _can_ force-update your lockfile by running `npm i --save babel-preset-expo@11.0.15` - and then removing this package again. npm's optimization should update the dependency chain `expo@51 -> babel-preset-expo@11.0.15` by doing so.
> 
> Sorry for the trouble!

Thanks Cedric.  `npx create-expo-app@latest` command works again.

**Comment by Asekon** [comment]
> Hey all! We've [issued an emergency hotfix](https://github.com/expo/expo/pull/31917) for `babel-preset-expo@11.0.15`. This pins the React Compiler version to an older but working one. We've also pinged Meta to get a proper fix for `babel-plugin-react-compiler` ASAP.
> 
> For now, it's likely that using `experiments.reactCompiler = true` will break your project. But if you don't use the React Compiler experiment, your project will install normally.
> 
> If you still encounter this, update your lock file to install `babel-preset-expo@11.0.15` for SDK 51, not `babel-preset-expo@<=11.0.14`. New projects created with `npx create-expo(-app)` should work again.
> 
> Important
> 
> `babel-preset-expo` is a dependency of `expo` and, preferably, should not be installed directly in your project. However, you _can_ force-update your lockfile by running `npm i --save babel-preset-expo@11.0.15` - and then removing this package again. npm's optimization should update the dependency chain `expo@51 -> babel-preset-expo@11.0.15` by doing so.
> 
> Sorry for the trouble!





Generating Pods project
Setting USE_HERMES build settings
Setting REACT_NATIVE build settings
[Ccache]: Removing Ccache from CC, LD, CXX & LDPLUSPLUS build settings
[Privacy Manifest Aggregation] No existing PrivacyInfo.xcprivacy file found, creating a new one.
[Privacy Manifest Aggregation] Reading .xcprivacy files to aggregate all used Required Reason APIs.
[!] An error occurred while processing the post-install hook of the Podfile.
undefined method `each' for nil:NilClass
/Users/expo/workingdir/build/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb:111:in `block (4 levels) in get_used_required_reason_apis'
/Users/expo/workingdir/build/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb:106:in `each'
/Users/expo/workingdir/build/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb:106:in `block (3 levels) in get_used_required_reason_apis'



i have been getting this log on the "install pods" step during EAS build for ios and been pulling my hair for the past day even though i had just built recently with no issues at all . Im assuming this is the source of the issue ?

**Comment by byCedric** [comment]
@Asekon No, this seems related to cocoapods, where this issue is specifically related to Node packages (npm, yarn, pnpm, bun). I would recommend opening an issue on [expo/eas-cli](https://github.com/expo/eas-cli) instead.

**Comment by jaslloyd** [comment]
> > Hey all! We've [issued an emergency hotfix](https://github.com/expo/expo/pull/31917) for `babel-preset-expo@11.0.15`. This pins the React Compiler version to an older but working one. We've also pinged Meta to get a proper fix for `babel-plugin-react-compiler` ASAP.
> > For now, it's likely that using `experiments.reactCompiler = true` will break your project. But if you don't use the React Compiler experiment, your project will install normally.
> > If you still encounter this, update your lock file to install `babel-preset-expo@11.0.15` for SDK 51, not `babel-preset-expo@<=11.0.14`. New projects created with `npx create-expo(-app)` should work again.
> > Important
> > `babel-preset-expo` is a dependency of `expo` and, preferably, should not be installed directly in your project. However, you _can_ force-update your lockfile by running `npm i --save babel-preset-expo@11.0.15` - and then removing this package again. npm's optimization should update the dependency chain `expo@51 -> babel-preset-expo@11.0.15` by doing so.
> > Sorry for the trouble!
> 
> ```
> Setting USE_HERMES build settings
> Setting REACT_NATIVE build settings
> [Ccache]: Removing Ccache from CC, LD, CXX & LDPLUSPLUS build settings
> [Privacy Manifest Aggregation] No existing PrivacyInfo.xcprivacy file found, creating a new one.
> [Privacy Manifest Aggregation] Reading .xcprivacy files to aggregate all used Required Reason APIs.
> [!] An error occurred while processing the post-install hook of the Podfile.
> undefined method `each' for nil:NilClass
> /Users/expo/workingdir/build/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb:111:in `block (4 levels) in get_used_required_reason_apis'
> /Users/expo/workingdir/build/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb:106:in `each'
> /Users/expo/workingdir/build/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb:106:in `block (3 levels) in get_used_required_reason_apis'
> ```
> 
> i have been getting this log on the "install pods" step during EAS build for ios and been pulling my hair for the past day even though i had just built recently with no issues at all . Im assuming this is the source of the issue ?

Our team had this error, if you run npx expo-doctor you should see a bunch of outdated packages...if you update them all it should pass.

**Comment by TowhidKashem** [comment]
> Hey all! We've [issued an emergency hotfix](https://github.com/expo/expo/pull/31917) for `babel-preset-expo@11.0.15`. This pins the React Compiler version to an older but working one. We've also pinged Meta to get a proper fix for `babel-plugin-react-compiler` ASAP.
> 
> For now, it's likely that using `experiments.reactCompiler = true` will break your project. But if you don't use the React Compiler experiment, your project will install normally.
> 
> If you still encounter this, update your lock file to install `babel-preset-expo@11.0.15` for SDK 51, not `babel-preset-expo@<=11.0.14`. New projects created with `npx create-expo(-app)` should work again.
> 
> Important
> 
> `babel-preset-expo` is a dependency of `expo` and, preferably, should not be installed directly in your project. However, you _can_ force-update your lockfile by running `npm i --save babel-preset-expo@11.0.15` - and then removing this package again. npm's optimization should update the dependency chain `expo@51 -> babel-preset-expo@11.0.15` by doing so.
> 
> Sorry for the trouble!

Hey @byCedric, just to confirm this means deploys in EAS will also work now? I only have 1 free iOS deploy left and want to make it count haha

**Comment by mlazari** [comment]
> what's crazy to me is Iv just had a successful build 3 weeks ago

@Asekon 2 possible reasons I can think of for that situation are:
1. You don't keep the package-lock.json / yarn.lock file in the repository / git ignore it and when you reinstalled the dependencies some of them changed.
2. You changed the version of node / cocoapods you're using and that somehow affected it (I think this is less likely than the first one though)

**Comment by Asekon** [comment]
> > what's crazy to me is Iv just had a successful build 3 weeks ago
> 
> @Asekon 2 possible reasons I can think of for that situation are:
> 
> 1. You don't keep the package-lock.json / yarn.lock file in the repository / git ignore it and when you reinstalled the dependencies some of them changed.
> 2. You changed the version of node / cocoapods you're using and that somehow affected it (I think this is less likely than the first one though)

I dont think its option 1 since I havent changed my gitignore or the way i build ( i keep the package-lock.json inside the project). However i did upgrade node right when I started debugging a separate issue. But i am using EAS build so how would that affect the building process ?

**Comment by mlazari** [comment]
@Asekon Not sure, it was just a guess, might be something else. Changing the node version locally would not affect the EAS build, the node version for that is configured in the `eas.json` file as far as I know.

**Comment by mlazari** [comment]
@Asekon This issue seems similar with yours. It looks like they fixed it by using a newer cocoapods version. The cocoapods version used by EAS build can be configured in the `eas.json` file:
- https://docs.expo.dev/build/eas-json/#selecting-build-tool-versions
- https://docs.expo.dev/eas/json/#cocoapods

**Comment by Asekon** [comment]
> @Asekon This issue seems similar with yours - [facebook/hermes#1410](https://github.com/facebook/hermes/issues/1410). It looks like they fixed it by using a newer cocoapods [version](https://github.com/CocoaPods/CocoaPods/releases). The cocoapods version used by EAS build can be configured in the `eas.json` file:
> 
> * https://docs.expo.dev/build/eas-json/#selecting-build-tool-versions
> * https://docs.expo.dev/eas/json/#cocoapods

Oh man i came across that issue before but i glossed over it for some reason. It worked, youre a life saver. May your blanket remain cold on both sides forever.
